Realm/RLMDictionary.h

```diff
@@ -0,0 +1,382 @@
+ */
+@property (nonatomic, readonly, assign) NSUInteger count;
+
+/**
+ The type of the value objects in the dictionary.
+ */
+@property (nonatomic, readonly, assign) RLMPropertyType type;
+
+/**
+ The type of the key object in the dictionary.
+ */
+@property (nonatomic, readonly, assign) RLMPropertyType keyType;
+
+/**
+ Indicates whether the objects in the collection can be `nil`.
+ */
+@property (nonatomic, readonly, getter = isOptional) BOOL optional;
+
+/**
+ The class name of the objects contained in the dictionary.
+
+ Will be `nil` if `type` is not RLMPropertyTypeObject.
+ */
+@property (nonatomic, readonly, copy, nullable) NSString *objectClassName;
+
+/**
+ The Realm which manages the dictionary. Returns `nil` for unmanaged dictionary.RLMDictionaryKey
+ */
+@property (nonatomic, readonly, nullable) RLMRealm *realm;
+
+/**
+ Indicates if the dictionary can no longer be accessed.
+ */
+@property (nonatomic, readonly, getter = isInvalidated) BOOL invalidated;
+
+/**
+ Indicates if the dictionary is frozen.
+
+ Frozen dictionaries are immutable and can be accessed from any thread. Frozen dictionaries
+ are created by calling `-freeze` on a managed live dictionary. Unmanaged dictionaries are
+ never frozen.
+ */
+@property (nonatomic, readonly, getter = isFrozen) BOOL frozen;
+
+#pragma mark - Accessing Objects from an Dictionary
+
+/**
+ Returns the value associated with a given key.
+
+ @param key The name of the property.
+
+ @return A value associated with a given key or `nil`.
+ */
+- (nullable id)valueForKey:(nonnull RLMKeyType <RLMDictionaryKey>)key;
+
+/**
+ Returns an object, if present, for a given key in the dictionary.
+ */
+- (nullable RLMObjectType)objectForKey:(nonnull RLMKeyType <RLMDictionaryKey>)key;
+
+/**
+ Returns an array containing the dictionary’s keys.
+ */
+@property(readonly, copy) NSArray<RLMKeyType <RLMDictionaryKey>> *allKeys;
+
+/**
+ Returns an array containing the dictionary’s values.
+ */
+@property(readonly, copy) NSArray<RLMObjectType> *allValues;
+
+/// :nodoc:
+- (nullable RLMObjectType)objectForKeyedSubscript:(RLMKeyType <RLMDictionaryKey>)key;
+
+/**
+ Applies a given block object to the each key-value pair of the dictionary
+ */
+- (void)enumerateKeysAndObjectsUsingBlock:(void (^)(RLMKeyType <RLMDictionaryKey> key, RLMObjectType obj, BOOL *stop))block;
+
+#pragma mark - Adding, Removing, and Replacing Objects in a Dictionary
+
+/**
+ Replace the data of a dictionary with the data of another dictionary.
+ */
+- (void)setDictionary:(RLMDictionary<RLMKeyType <RLMDictionaryKey>, RLMObjectType> *)otherDictionary;
+
+/**
+ Delete all dictionary's keys and values.
+ */
+- (void)removeAllObjects;
+
+/**
+ Adds an array of distinct objects to the set.
+
+ @warning This method may only be called during a write transaction.
+
+ @param objects      `NSDictionary` which contains objects of the
+                    same class as the dictionary.
+ */
+- (void)addObjects:(NSDictionary *)objects;
```

Comment:
This is `addEntriesFromDictionary:`.

Reply:
it is used in RLMAccessor's unmanagedSetter(RLMProperty *prop, const char *). addEntriesFromDictionary is defined on it's own.